### PR TITLE
[RA-1 CH04] 4.2.3.3. Load Balancing

### LINKED ISSUE #725: [RA-1 Ch04] 4.2.3.3. LBaaS v2 compliant Load Balancing
[RA-1 Ch04] 4.2.3.3. LBaaS v2 compliant Load Balancing  -- LBaaS v2 is in the process of being deprecated -- replaced by Octavia.

Suggest change of Title of 4.2.3.3 to Octavia Load Balancer; and the following content:

Load balancing is needed for automatic scaling and availability.  [Octavia]( https://docs.openstack.org/octavia/latest/reference/introduction.html) is an open-source load balancer for OpenStack and replaces the deprecated (as of OpenStack Queens release) Neutron LBaaS. Octavia supports the Neutron LBaaS v2 API and also has a similar CLI for seamless transition. 

Octavia delivers horizontal scaling by managing a fleet of VMs, containers or bare metal servers (collectively amphorae) and spinning up the necessary resources on demand. To accomplish this Octavia depends upon a number of OpenStack services including Nova for spinning up compute resources on demand and their life cycle management; Neutron for connectivity between the compute resources, project environment and external networks; Keystone for authentication; and Glance for storing of the compute resource images.

### PR DESCRIPTION
**Pull request by pgoyal01**
[RA-1 CH04] 4.2.3.3. Octavia v2 API compliant Load Balancing

Fixes #725

### CODE REVIEW

File: doc/ref_arch/openstack/chapters/chapter04.md [hunk 1]
```diff
@@ -164,8 +164,15 @@ Caveats:
...
 
-#### 4.2.3.3. LBaaS v2 compliant Load Balancing
-**Content to be developed**
+#### 4.2.3.3. Octavia v2 API compliant Load Balancing
+Load balancing is needed for automatic scaling, managing availability and changes. [Octavia](https://docs.openstack.org/octavia/latest/reference/introduction.html) is an open-source load balancer for OpenStack, based on HAProxy, and replaces the deprecated (as of OpenStack Queens release) Neutron LBaaS. The Octavia v2 API is a superset of the deprecated Neutron LBaaS v2 API and has a similar CLI for seamless transition. 
+
+As a default Octavia utilizes Amphorae Load Balancer. Amphorae consists of a fleet of VMs, containers or bare metal servers and delivers horizontal scaling by managing and spinning these resources on demand. The reference implementation of the Amphorae image is an Ubuntu virtual machine running HAProxy. 
+
+Octavia depends upon a number of OpenStack services including Nova for spinning up compute resources on demand and their life cycle management; Neutron for connectivity between the compute resources, project environment and external networks; Keystone for authentication; and Glance for storing of the compute resource images.
```
Comment (deebhatia) [inline doc/ref_arch/openstack/chapters/chapter04.md]: Skipping Barbican for TLS management is intentional?

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter04.md]: @deebhatia Not trying to get into all of the Octavia interactions. But if they are important then could add a sentence.

File: doc/ref_arch/openstack/chapters/chapter04.md [hunk 2]
```diff
@@ -164,8 +164,15 @@ Caveats:
...
+
+Octavia supports provider drivers which allows third party load balancing drivers (such as F5, AVI, ...) to be utilized instead of the default Amphorae load balancer. When creating a third-party load balancer, the **provider** attribute is used to specify the backend to be used to create the load balancer. The **list providers** lists all enabled provider drivers.  Instead of using the provider parameter, an alternate is to specify the flavor_id in the create call where provider-specific Octavia flavors have been created.
```
[the remark this answers is not in the document]
Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter04.md]: @msidana Done - Thanks

File: doc/ref_arch/openstack/chapters/chapter04.md (hunk 1 above)
Comment (deebhatia) [reply doc/ref_arch/openstack/chapters/chapter04.md]: Okay.

**Comment by rabiabdel** [comment]
@iangardner22 please review

**Pull request merged**